### PR TITLE
Don't register edges twice in merge return

### LINKED ISSUE #3267: Merge return produces a phi with too many entries
```
               OpCapability Shader
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint Fragment %2 "main" %3
               OpExecutionMode %2 OriginUpperLeft
               OpDecorate %4 RelaxedPrecision
               OpDecorate %3 Location 0
               OpMemberDecorate %_struct_5 0 Offset 0
               OpDecorate %_struct_5 Block
               OpDecorate %6 DescriptorSet 0
               OpDecorate %6 Binding 0
               OpDecorate %7 RelaxedPrecision
               OpDecorate %8 RelaxedPrecision
               OpDecorate %9 RelaxedPrecision
               OpDecorate %10 RelaxedPrecision
               OpDecorate %11 RelaxedPrecision
               OpDecorate %10 RelaxedPrecision
               OpDecorate %10 RelaxedPrecision
       %void = OpTypeVoid
         %13 = OpTypeFunction %void
        %int = OpTypeInt 32 1
         %15 = OpTypeFunction %int
      %float = OpTypeFloat 32
    %v4float = OpTypeVector %float 4
%_ptr_Output_v4float = OpTypePointer Output %v4float
          %3 = OpVariable %_ptr_Output_v4float Output
    %float_1 = OpConstant %float 1
    %float_0 = OpConstant %float 0
         %21 = OpConstantComposite %v4float %float_1 %float_0 %float_0 %float_1
    %v2float = OpTypeVector %float 2
  %_struct_5 = OpTypeStruct %v2float
%_ptr_Uniform__struct_5 = OpTypePointer Uniform %_struct_5
          %6 = OpVariable %_ptr_Uniform__struct_5 Uniform
      %int_0 = OpConstant %int 0
       %uint = OpTypeInt 32 0
     %uint_1 = OpConstant %uint 1
%_ptr_Uniform_float = OpTypePointer Uniform %float
       %bool = OpTypeBool
%_ptr_Function_int = OpTypePointer Function %int
      %int_1 = OpConstant %int 1
      %false = OpConstantFalse %bool
         %32 = OpUndef %int
       %true = OpConstantTrue %bool
          %2 = OpFunction %void None %13
         %34 = OpLabel
          %9 = OpFunctionCall %int %4
               OpReturn
               OpFunctionEnd
          %4 = OpFunction %int None %15
         %35 = OpLabel
          %7 = OpVariable %_ptr_Function_int Function
          %8 = OpVariable %_ptr_Function_int Function
               OpStore %3 %21
               OpBranch %36
         %36 = OpLabel
         %10 = OpPhi %int %32 %35 %11 %37
               OpLoopMerge %38 %37 None
               OpBranch %39
         %39 = OpLabel
         %40 = OpAccessChain %_ptr_Uniform_float %6 %int_0 %uint_1
         %41 = OpLoad %float %40
         %42 = OpFOrdLessThan %bool %41 %float_0
               OpSelectionMerge %43 None
               OpBranchConditional %42 %44 %45
         %45 = OpLabel
               OpStore %7 %int_0
               OpBranch %46
         %46 = OpLabel
         %47 = OpSLessThan %bool %int_0 %int_1
               OpLoopMerge %48 %49 None
               OpBranch %50
         %50 = OpLabel
               OpSelectionMerge %51 None
               OpBranchConditional %42 %48 %51
         %51 = OpLabel
               OpReturnValue %int_1
         %49 = OpLabel
               OpBranch %46
         %48 = OpLabel
               OpSelectionMerge %52 None
               OpBranchConditional %42 %53 %52
         %53 = OpLabel
               OpBranch %54
         %54 = OpLabel
               OpLoopMerge %55 %56 None
               OpBranch %57
         %57 = OpLabel
               OpReturnValue %int_1
         %56 = OpLabel
               OpBranch %54
         %55 = OpLabel
               OpUnreachable
         %52 = OpLabel
               OpBranch %37
         %44 = OpLabel
               OpBranch %37
         %43 = OpLabel
               OpUnreachable
         %37 = OpLabel
         %11 = OpPhi %int %10 %44 %10 %52
               OpBranchConditional %false %36 %38
         %38 = OpLabel
               OpReturnValue %11
               OpFunctionEnd
```

Run `--merge-return` and validate the result. Should see the following error:

>error: line 125: OpPhi's number of incoming blocks (6) does not match block's predecessor count (5).
>  %72 = OpPhi %int %71 %48 %71 %43 %11 %37 %71 %69 %71 %55 %71 %69

### PR DESCRIPTION
**Pull request by s-perron**
Fixes #3267

### CODE REVIEW

File: test/opt/pass_merge_return_test.cpp [hunk 1]
```diff
@@ -2005,6 +2005,78 @@ TEST_F(MergeReturnPassTest, SingleReturnInMiddle) {
   SinglePassRunAndMatch<MergeReturnPass>(before, false);
 }
 
+TEST_F(MergeReturnPassTest, PhiWithTooManyEntries) {
+  // Check that the OpPhi node has the correct number of entries.  This is check
```
Comment (kuhar) [inline test/opt/pass_merge_return_test.cpp:2009]: nit: `this is checked`?

File: test/opt/pass_merge_return_test.cpp [hunk 2]
```diff
@@ -2005,6 +2005,78 @@ TEST_F(MergeReturnPassTest, SingleReturnInMiddle) {
...
+               OpBranch %31
+         %31 = OpLabel
+               OpReturnValue %int_1
+         %30 = OpLabel
+               OpBranch %28
+         %29 = OpLabel
+               OpUnreachable
+         %26 = OpLabel
+               OpBranch %17
+         %17 = OpLabel
```
Comment (kuhar) [inline test/opt/pass_merge_return_test.cpp:2069]: Would it be possible to make this test shorter?

Comment (s-perron) [reply test/opt/pass_merge_return_test.cpp:2069]: Not is it not.  I tried removing everything I could while still reproducing the original error and this is the smallest I could make it.

Comment (kuhar) [reply test/opt/pass_merge_return_test.cpp:2069]: OK

File: test/opt/pass_merge_return_test.cpp (hunk 1 above)
Comment (s-perron) [reply test/opt/pass_merge_return_test.cpp]: Done.

**Pull request merged**